mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/operation/OperationResult.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.operation;
+
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.api.metadata.MediaType;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Represents the result of an operation execution. Extensions can use this class
+ * for cases in which the operation not only needs to return a value to be set
+ * on the message payload but also wants to specify message attributes and/or
+ * {@link MediaType}.
+ * <p>
+ * The {@link #getOutput()} value is always taken at face value, meaning that
```

Comment:
this is already how it works. For void operations check `VoidReturnDelegate`. As for what you say about the runtime deciding where the output goes, that's also how it works, I just phrased it wrong because I forgot that the output can go in the target. I fixed that